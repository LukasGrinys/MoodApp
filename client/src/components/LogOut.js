import React, {Component} from 'react';
import Loading from './Loading/loading';
import { connect } from 'react-redux';
import { logOut, clearLogs } from './../actions';

class LogOut extends Component {
    UNSAFE_componentWillMount(){
        this.props.dispatch(logOut());
    }
    componentWillUnmount(){
        this.props.dispatch(clearLogs());
    }
    render(){
        return (
            <div>
                <Loading nightmode={this.props.nightmode}/>
            </div>
        );
    }
};

function mapStateToProps(state) {
    return {
        user: state.user,
        logs: state.logs
    }
}
export default connect(mapStateToProps)(LogOut);