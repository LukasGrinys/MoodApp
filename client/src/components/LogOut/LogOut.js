import React, { useEffect } from 'react';
import Loading from '../Loading/loading';
import { logOut } from '../../actions/user/asyncActions';
import { useDispatch } from 'react-redux';

const LogOut = () => {
    const dispatch = useDispatch();

    useEffect( () => {
        dispatch(logOut());
    }, [])

    return <Loading/>
};

export default LogOut;