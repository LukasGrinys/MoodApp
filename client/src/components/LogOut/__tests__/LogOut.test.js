import React from 'react';
import { Provider } from 'react-redux';
import LogOut from '../LogOut';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
// import { logOut } from '../../../actions/user/asyncActions';

const mockStore = configureStore();
const store = mockStore({

});

jest.mock("react-redux", () => {  
    const { Provider } = jest.requireActual("react-redux");

    return {
        Provider,
        useDispatch: () => jest.fn(),
    };
});

const wrapper = mount(
    <Provider store={store}>
        <LogOut/>
    </Provider>
);

test('Renders properly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Shows Loading widget', () => {
    expect(wrapper.find('Loading').length).toBe(1);
});

test('Dispatches logOut action', () => {
    const logOut = jest.fn();
    wrapper.update();

    setTimeout( () => {
        expect(logOut).toHaveBeenCalledTimes(1);
        done();
    }, 0);
})


