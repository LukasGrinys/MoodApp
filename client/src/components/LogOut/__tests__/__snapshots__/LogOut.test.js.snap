// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LogOut>
    <Loading>
      <div
        className="loadingScreen"
      >
        <div>
          Loading
        </div>
        <br />
        <div
          className="barContainer"
        >
          <div
            className="bar"
          />
          <div
            className="bar delay1"
          />
          <div
            className="bar delay2"
          />
        </div>
      </div>
    </Loading>
  </LogOut>
</Provider>
`;
